### PR TITLE
shapes: v2: add SHACL shapes for new RML ontology

### LINKED ISSUE #17: Testcases: add tests
Testcases are here: https://github.com/kg-construct/rml-star-test-cases

- Do they cover the current spec?
- Do they include invalid cases, thus throwing errors?

Useful for validating the SHACL shapes and make sure we cover everything

### PR DESCRIPTION
**Pull request by DylanVanAssche**
These shapes covers the RML-Star specification:

- rmls:NonAssertedTriplesMap
- rmls:StarMap

The generate_shapes.py script combines these shapes into a single shape for SHACL validators 'star.ttl'

Fixes #17

### CODE REVIEW

**Comment by DylanVanAssche** [comment]
The `sh:description` entries need work, please provide some feedback what they should be

File: shapes/v2/nonassertedtriples_map.ttl [hunk 1]
```diff
@@ -0,0 +1,100 @@
+###############################################################################
+# RMLS Non Asserted Triples Map shape                                         #
+# Copyright Dylan Van Assche, IDLab - UGent - imec (2020 - 2023)              #
+###############################################################################
+@prefix schema: <http://schema.org/> .
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rml: <http://w3id.org/rml/core/> .
+@prefix rmls: <http://w3id.org/rml/star/> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
```
Comment (anaigmo) [inline shapes/v2/nonassertedtriples_map.ttl]: update to:
```
@prefix rml: <http://w3id.org/rml/> .
```
the `rmls` prefix is not needed

Comment (anaigmo) [reply shapes/v2/nonassertedtriples_map.ttl]: the `rmls` prefix applies to the entire document, it'd need to be replaced with `rml`

File: shapes/v2/nonassertedtriples_map.ttl [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+schema:RMLSNonAssertedTriplesMapShape
+    a sh:NodeShape ;
+    sh:targetClass rmls:NonAssertedTriplesMap ;
+    sh:message "rmls:NonAssertedTriplesMap" ;
+    sh:description """
+    Represents a non asserted triples map.
+    """ ;
+    sh:message """
+    rmls:NonAssertedTriplesMap requires exactly one rml:subject or 
+    one rml:subjectMap and zero or more rml:predicateObjectMaps.
```
Comment (anaigmo) [inline shapes/v2/nonassertedtriples_map.ttl]: ...and exactly one logical source (according also to the restrictions below)

File: shapes/v2/star.ttl [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    sh:targetClass <http://w3id.org/rml/star/NonAssertedTriplesMap> .
+
+<http://schema.org/RMLSStarMapShape> a sh:NodeShape ;
+    sh:description """
+    Represents a star map.
+    """ ;
+    sh:ignoredProperties ( rdf:type ) ;
+    sh:message """
+    rmls:StarMap requires exactly one rml:subject or one rml:subjectMap and zero
+    or more rml:predicateObjectMaps
```
Comment (anaigmo) [inline shapes/v2/star.ttl]: This desc is for the NonAssertedTriplesMap, please replace with "rml:StarMap requires exactly one rml:quotedTriplesMap and zero or more rml:joinCondition."

File: shapes/v2/star.ttl [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            sh:message """
+        Exactly one rmls:quotedTriplesMap is required.
+        """ ;
+            sh:minCount 1 ;
+            sh:name "rmls:quotedTriplesMap" ;
+            sh:nodeKind sh:BlankNodeOrIRI ;
+            sh:path <http://w3id.org/rml/star/quotedTriplesMap> ],
+        [ sh:description """
+        Specifies the join condition for joining the child logical table with 
+        the parent logical table of the foreign key constraint.
```
Comment (anaigmo) [inline shapes/v2/star.ttl]: Please replace with the new description of join condition: "Specifies the join condition for joining the child logical source with the parent logical source of the foreign key constraint."

File: shapes/v2/star.ttl [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        """ ;
+            sh:message """
+        rml:joinCondition must specify an rml:parent and rml:child.
+        """ ;
+            sh:name "rml:joinCondition" ;
+            sh:nodeKind sh:BlankNodeOrIRI ;
+            sh:path rml:joinCondition ;
```
Comment (anaigmo) [inline shapes/v2/star.ttl]: I see cardinality restrictions for `child` and `parent`, but not for joinCondition, which would be the same as for the core spec (0..N I think)

Also, the core spec has added `childMap` and `parentMap`, according to [#59](https://github.com/kg-construct/rml-core/issues/59), which may be need to be reflected here

File: shapes/v2/star_map.ttl [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+schema:RMLSStarMapShape
+    a sh:NodeShape ;
+    sh:targetClass rmls:StarMap ;
+    sh:message "rmls:StarMap" ;
+    sh:description """
+    Represents a star map.
+    """ ;
+    sh:message """
+    rmls:StarMap requires exactly one rml:subject or one rml:subjectMap and zero
+    or more rml:predicateObjectMaps
```
Comment (anaigmo) [inline shapes/v2/star_map.ttl]: same comment as in the star.ttl file regarding the desc of the starmap

**Review by anaigmo — CHANGES_REQUESTED** [review]
They look great :) I've made some comments regarding some descriptions and for changing the IRIs.

I'm missing some restrictions regarding the range of the core properties `rml:subjectMap` and `rml:objectMap` (`rml:StarMap`). In the core, we strip all references to this module, so I think they would be put here instead if there is a way that they not enter in conflict with the core ones

**Comment by DylanVanAssche** [comment]
@anaigmo We reworked the shapes in Core to have inheritance for things like `joinCondition`, etc. which are shared among multiple resources. Once merged in Core, I can update them here to match.

**Pull request merged**